From 08053be7adc6d4a83191efed735c3f565bcbcf1f Mon Sep 17 00:00:00 2001
From: Samuli Suominen <ssuominen@gentoo.org>
Date: Sat, 5 May 2012 17:13:45 +0200
Subject: [PATCH] Fix panel plugin build.

- Don't use a versioned library.
- Build as a module.
- Only export needed symbols.
---
Upstream-Status: applied
 Makefile.am |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 45100fd..d0c0adb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -132,13 +132,19 @@ panel_plugin_libscreenshooterplugin_la_CFLAGS =	\
 	@SOUP_CFLAGS@
 
 panel_plugin_libscreenshooterplugin_la_LDFLAGS = \
+	-avoid-version \
+	-module \
+	-no-undefined \
+	-export-symbols-regex '^xfce_panel_module_(preinit|init|construct)' \
+	$(PLATFORM_LDFLAGS)
+
+panel_plugin_libscreenshooterplugin_la_LIBADD = \
 	@EXO_LIBS@ \
 	@LIBXFCE4PANEL_LIBS@ \
 	@GTHREAD_LIBS@ \
 	@SOUP_LIBS@ \
-	@LIBXFCE4UI_LIBS@
-
-panel_plugin_libscreenshooterplugin_la_LIBADD = lib/libscreenshooter.la
+	@LIBXFCE4UI_LIBS@ \
+	lib/libscreenshooter.la
 
 panel_plugin_libscreenshooterplugin_la_SOURCES = \
 	panel-plugin/screenshooter-plugin.c
-- 
1.7.4.4

